db/migrate/20240903024311_backfill_user_passwords_from_users.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+class BackfillUserPasswordsFromUsers < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+  BATCH_SIZE = 100_000
+
+  def up
+    min_result = DB.query(<<~SQL.squish)
+        SELECT id
+        FROM users
+        WHERE password_hash IS NOT NULL
+        ORDER BY id ASC
+        LIMIT 1;
+    SQL
+    last_queried_id = min_result&.first&.id # we have bot users set with negative ids;
```

Comment:
Not critical, but you might be interested in `DB.query_single`, which would let you remove the `&.first`